test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,423 @@
+            {
+                result.Channel().Close(); // Avoids using cached channel everytime the test runs
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                channelOperationResult = result.ExtendedError();
+            }
+
+            SetEvent(channelEvent.get());
+        });
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent.get(), 960000 /* milliseconds */))
+    {
+        channelOperation.Cancel();
+        channelOperationResult = ERROR_TIMEOUT;
+    }
+    else
+    {
+        channelOperation.Close(); // Do not call getresults after this
+    }
+
+    return (channelOperationResult == S_OK);
+}
+
+bool MultipleChannelRequestUsingSameRemoteId()
+{
+    wil::unique_handle channelEvent1 = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+    wil::unique_handle channelEvent2 = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+
+    auto channelOperationResult2 = S_OK;
+
+    auto channelOperation1 = PushNotificationManager::CreateChannelAsync(remoteId1);
+
+    channelOperation1.Completed(
+        [&channelEvent1](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                result.Channel().Close(); // Avoids using cached channel everytime the test runs
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+            }
+
+            SetEvent(channelEvent1.get());
+        });
+
+    auto channelOperation2 = PushNotificationManager::CreateChannelAsync(remoteId1);
+
+    channelOperation2.Completed(
+        [&channelEvent2, &channelOperationResult2](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                result.Channel().Close(); // Avoids using cached channel everytime the test runs
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                channelOperationResult2 = result.ExtendedError();
+            }
+
+            SetEvent(channelEvent2.get());
+        });
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent2.get(), 960000 /* milliseconds */))
```

Comment:
Same problems as before. Switch to `wait_for` to solve a lot of problems (such as lifetime issues).

Reply:
@oldnewthing raymond can you point me to the API you are referring to?